core-test/src/test/scala/org/bitcoins/core/wallet/signer/SignerIntegrationTest.scala

```diff
@@ -0,0 +1,189 @@
+}
+import org.bitcoins.testkit.util.BitcoinSAsyncTest
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class SignerIntegrationTest extends BitcoinSAsyncTest {
+
+  implicit val ec: ExecutionContext = ExecutionContext.global
+
+  implicit override val generatorDrivenConfig: PropertyCheckConfiguration =
+    generatorDrivenConfigNewCode
+
+  behavior of "Signer"
+
+  def inputIndex(spendingInfo: UTXOSpendingInfo, tx: Transaction): Int = {
+    tx.inputs.zipWithIndex
+      .find(_._1.previousOutput == spendingInfo.outPoint) match {
+      case Some((_, index)) => index
+      case None =>
+        throw new IllegalArgumentException(
+          "Transaction did not contain expected input.")
+    }
+  }
+
+  def createProgram(
+      tx: Transaction,
+      idx: Int,
+      utxo: UTXOSpendingInfo): PreExecutionScriptProgram = {
+    val output = utxo.output
+
+    val spk = output.scriptPubKey
+
+    val amount = output.value
+
+    val txSigComponent = spk match {
+      case witSPK: WitnessScriptPubKeyV0 =>
+        val o = TransactionOutput(amount, witSPK)
+        WitnessTxSigComponentRaw(tx.asInstanceOf[WitnessTransaction],
+                                 UInt32(idx),
+                                 o,
+                                 Policy.standardFlags)
+      case _: UnassignedWitnessScriptPubKey => ???
+      case x @ (_: P2PKScriptPubKey | _: P2PKHScriptPubKey |
+          _: P2PKWithTimeoutScriptPubKey | _: MultiSignatureScriptPubKey |
+          _: WitnessCommitment | _: CSVScriptPubKey | _: CLTVScriptPubKey |
+          _: ConditionalScriptPubKey | _: NonStandardScriptPubKey |
+          EmptyScriptPubKey) =>
+        val o = TransactionOutput(CurrencyUnits.zero, x)
+        BaseTxSigComponent(tx, UInt32(idx), o, Policy.standardFlags)
+
+      case _: P2SHScriptPubKey =>
+        val p2shScriptSig =
+          tx.inputs(idx).scriptSignature.asInstanceOf[P2SHScriptSignature]
+        p2shScriptSig.redeemScript match {
+
+          case _: WitnessScriptPubKey =>
+            WitnessTxSigComponentP2SH(transaction =
+                                        tx.asInstanceOf[WitnessTransaction],
+                                      inputIndex = UInt32(idx),
+                                      output = output,
+                                      flags = Policy.standardFlags)
+
+          case _ =>
+            BaseTxSigComponent(tx, UInt32(idx), output, Policy.standardFlags)
+        }
+    }
+
+    PreExecutionScriptProgram(txSigComponent)
+  }
+
+  def verifyScripts(
+      tx: Transaction,
+      utxos: Vector[UTXOSpendingInfo]): Boolean = {
+    val programs: Vector[PreExecutionScriptProgram] =
+      tx.inputs.zipWithIndex.toVector.map {
+        case (input: TransactionInput, idx: Int) =>
+          val utxo = utxos.find(_.outPoint == input.previousOutput).get
+          createProgram(tx, idx, utxo)
+      }
+    ScriptInterpreter.runAllVerify(programs)
+  }
+
+  it must "sign inputs correctly" in {
```

Comment:
What is the difference between this and the test at the bottom of `BitcoinTxBuilderTest`?